Signal/src/ViewControllers/DebugUI/DebugUIDiskUsage.m

```diff
@@ -0,0 +1,299 @@
+                                                      = (TSAttachmentStream *)attachment;
+                                                  [attachmentStreams addObject:attachmentStream];
+                                              }];
+
+        DDLogInfo(@"Saving %zd attachment streams.", attachmentStreams.count);
+
+        // Persist the new localRelativeFilePath property of TSAttachmentStream.
+        // For performance, we want to upgrade all existing attachment streams in
+        // a single transaction.
+        for (TSAttachmentStream *attachmentStream in attachmentStreams) {
+            [attachmentStream saveWithTransaction:transaction];
+        }
+    }];
+}
+
++ (void)deleteOldMessages_3Months
+{
+    NSTimeInterval kMinute = 60.f;
+    NSTimeInterval kHour = 60 * kMinute;
+    NSTimeInterval kDay = 24 * kHour;
+    NSTimeInterval kMonth = 30 * kDay;
+    [self deleteOldMessages:kMonth * 3];
+}
+
++ (void)deleteOldMessages:(NSTimeInterval)maxAgeSeconds
+{
+    TSStorageManager *storageManager = [TSStorageManager sharedManager];
+    [storageManager.newDatabaseConnection readWriteWithBlock:^(YapDatabaseReadWriteTransaction *_Nonnull transaction) {
+
+        NSMutableArray<NSString *> *threadIds = [NSMutableArray new];
+        YapDatabaseViewTransaction *interactionsByThread = [transaction ext:TSMessageDatabaseViewExtensionName];
+        [interactionsByThread enumerateGroupsUsingBlock:^(NSString *group, BOOL *stop) {
+            [threadIds addObject:group];
+        }];
+        NSMutableArray<TSInteraction *> *interactionsToDelete = [NSMutableArray new];
+        for (NSString *threadId in threadIds) {
+            [interactionsByThread enumerateKeysAndObjectsInGroup:threadId
+                                                      usingBlock:^(NSString *collection,
+                                                          NSString *key,
+                                                          TSInteraction *interaction,
+                                                          NSUInteger index,
+                                                          BOOL *stop) {
+                                                          NSTimeInterval ageSeconds
+                                                              = fabs(interaction.dateForSorting.timeIntervalSinceNow);
+                                                          if (ageSeconds < maxAgeSeconds) {
+                                                              *stop = YES;
+                                                              return;
+                                                          }
+                                                          [interactionsToDelete addObject:interaction];
+                                                      }];
+        }
+
+        DDLogInfo(@"Deleting %zd interactions.", interactionsToDelete.count);
+
+        for (TSInteraction *interaction in interactionsToDelete) {
```

Comment:
If this ever becomes public facing we may want to batch this to avoid loading a billion interactions into memory at once.

I think it's completely reasonable as is for debug though.